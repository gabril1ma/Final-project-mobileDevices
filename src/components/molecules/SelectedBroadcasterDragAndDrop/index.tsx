import React from "react";
import { StyleSheet, Text, TouchableOpacity, View } from "react-native";
import useAppContext from "../../../hooks/useAppContext";

const SelectedBroadcasterDragAndDrop = () => {
  const {
    appState: { broadcasters },
  } = useAppContext();

  const selectedBroadCasters = broadcasters.filter((b) => b.selected);

  return (
    <View style={styles.container}>
      <Text style={styles.text}>
        Coloque em ordem crescente qual veiculo de noticia vocÃª prefere:
      </Text>

      <View style={styles.buttonContainer}>
        {selectedBroadCasters.map((b) => (
          <TouchableOpacity style={styles.draggableItem} key={b.name}>
            <Text>{b.name}</Text>
          </TouchableOpacity>
        ))}
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    marginTop: 16,
  },
  buttonContainer: {
    display: "flex",
    gap: 4,
    flexDirection: "row",
    flexWrap: "wrap",
  },
  text: {
    color: "#18224C",
    fontWeight: "500",
    fontSize: 15,
    marginBottom: 8,
  },
  draggableItem: {
    padding: 16,
    color: "#18224C",
    backgroundColor: "#DFE4EA",
    fontWeight: "400",
    fontSize: 13,
    fontFamily:
      " -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica, Arial, sans-serif",
  },
});

export default SelectedBroadcasterDragAndDrop;
