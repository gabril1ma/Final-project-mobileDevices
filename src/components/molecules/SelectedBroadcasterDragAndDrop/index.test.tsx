import { create, ReactTestRendererJSON } from "react-test-renderer";
import SelectedBroadcasterDragAndDrop from ".";

export type TTreeRenderer = ReactTestRendererJSON | ReactTestRendererJSON[];

const broadcasterMock = [
  {
    name: "CNN Brasil",
    selected: true,
    urlAPI: "",
  },
  {
    name: "Folha de SÃ£o Paulo",
    selected: true,
    urlAPI: "",
  },
  {
    name: "Estado de Minas",
    selected: true,
    urlAPI: "",
  },
];

jest.mock("../../../hooks/useAppContext", () => ({
  __esModule: true,
  default: () => ({
    appState: {
      broadcasters: broadcasterMock,
    },
    setAppState: () => jest.fn(),
  }),
}));

describe("<SelectedBroadcasterDragAndDrop />", () => {
  let tree: TTreeRenderer;
  let testTree: ReactTestRendererJSON;

  beforeEach(() => {
    tree = create(<SelectedBroadcasterDragAndDrop />).toJSON();
    testTree = tree[0] ?? tree;
  });

  it("should render component", () => {
    expect(testTree).toMatchSnapshot();
  });

  it("should have one children", () => {
    expect(testTree?.children?.length).toBe(2);
  });
});
