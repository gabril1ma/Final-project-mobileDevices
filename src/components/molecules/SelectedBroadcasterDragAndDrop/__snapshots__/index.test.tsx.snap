// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectedBroadcasterDragAndDrop /> should render component 1`] = `
<View
  style={
    {
      "marginTop": 16,
    }
  }
>
  <Text
    style={
      {
        "color": "#18224C",
        "fontSize": 15,
        "fontWeight": "500",
        "marginBottom": 8,
      }
    }
  >
    Coloque em ordem crescente qual veiculo de noticia você prefere:
  </Text>
  <View
    style={
      {
        "display": "flex",
        "flexDirection": "row",
        "flexWrap": "wrap",
        "gap": 4,
      }
    }
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#DFE4EA",
          "color": "#18224C",
          "fontFamily": " -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica, Arial, sans-serif",
          "fontSize": 13,
          "fontWeight": "400",
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      CNN Brasil
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#DFE4EA",
          "color": "#18224C",
          "fontFamily": " -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica, Arial, sans-serif",
          "fontSize": 13,
          "fontWeight": "400",
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      Folha de São Paulo
    </View>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "backgroundColor": "#DFE4EA",
          "color": "#18224C",
          "fontFamily": " -apple-system, BlinkMacSystemFont, Segoe UI, Roboto, Helvetica, Arial, sans-serif",
          "fontSize": 13,
          "fontWeight": "400",
          "opacity": 1,
          "padding": 16,
        }
      }
    >
      Estado de Minas
    </View>
  </View>
</View>
`;
